tensorrt_llm/_torch/visual_gen/profiler.py

```
@@ -0,0 +1,314 @@
+ProfileRange = Union[str, Tuple[frozenset, frozenset], None]
+
+
+def parse_profile_range() -> ProfileRange:
+    """Parse ``TLLM_PROFILE_VISUAL_GEN_START_STOP`` for profiler scoping.
+
+    Visual-gen-specific env var (separate from the LLM path's
+    ``TLLM_PROFILE_START_STOP``). Use with ``nsys profile -c cudaProfilerApi ...``.
+
+    Supported formats:
+
+    * ``A-B``            – profile denoise steps A through B
```

Ruff (0.16.0)

[warning] Docstring contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)